### PR TITLE
Fix #24

### LINKED ISSUE #24: Css selector add attr not work correctly
1.I write code `content = Css('.download_button', 'href')`  in Class Post but not work.
Error info : 

>  Selector ".video-download-button" for url was wrong, please check again

which means that value is None.In fact,execute code `d(self.rule)[0].attr(self.attr, None)`  would terminate and output has not attr attribute...

2.Now selector just select the first element since the code `d(self.rule)[0].text` 。How can i choose the whole elements match self.rule and acquire their attr? (I searched the doc http://pyquery.readthedocs.io/en/latest/, but not found answer)

### PR DESCRIPTION
**Pull request by Jie-OY**
Fix issue #24 .Enable select all element matches self.rule and acquire the self.attr
attribute value

### CODE REVIEW

File: gain/selector.py [hunk 1]
```diff
@@ -23,11 +23,15 @@ class Css(Selector):
     def parse_detail(self, html):
         d = pq(html)
         if self.attr is None:
-            try:
-                return d(self.rule)[0].text
-            except IndexError:
-                return None
-        return d(self.rule)[0].attr(self.attr, None)
+            rlt = []
```
Comment (Jie-OY) [inline gain/selector.py]: Yep,i was thinking about what to return  at that time too.

At last, to compatiable with Usage Examplt code :
```
async def save(self):
        async with aiofiles.open('scrapinghub.txt', 'a+') as f:
            await f.write(self.results['title'])
```
I let func returns a str type value.

So,what's your opinion,return the rlt variable or something else?

Comment (gaojiuli) [inline gain/selector.py]: Your solution works well. But how to divide those string fields?

```python
class Book(Item):
    title=Css('.title')
    author=Css('.author')
```
A page with multiple `Book`. Your solution is 

```
Book(title='title1 title2 title3',author='author1 author2 author3')
```

What I want is 

```
Book(title='title1',author='author1'),
Book(title='title2',author='author1'),
Book(title='title3',author='author1')
```

Comment (Jie-OY) [inline gain/selector.py]: My humble opinion: Doing that is a little werid...For example,what if a page contains 10 title and 1 author?
As for problem how to divide those string fields,for the current commit version,we can use split().But that's werid,too. :)
So,now,I think return a list type value is more suitable.

`self.results.[title]` is a list represents all matched elements which length is 1 ,bigger than 1 or even 0.And so do `self.results.[author]`. Do you think this make sense?

Comment (gaojiuli) [inline gain/selector.py]: I have a better solution for the page with 1 author and 10 titles, just defined a model like:
```
class Book(Item):
    title=Css('.title',mutiple=true)
    author=Css('.author')
```
This return a list like what you said.

And for the page with 10 authors and 10 titles:
```
class Book(Item):
    __mutiple__=true
    title=Css('.title')
    author=Css('.author')
```
This works as I mentioned above.
What do you think?

Comment (Jie-OY) [inline gain/selector.py]: uh..First of all,I want to know for what purpose you are trying to generate many Item objects in one page :expressionless:.

What we ,maybe i, want is to handle each data crawled from web pages,isn't it?And the first way works fine with this purpose.

Besides,What if  every page matched one parser's URL rule contains different amount title and author.For example.one page has 5 title and 1 author,another has 1 title and 1 author.(And that's what I really encountered :joy:)

Comment (gaojiuli) [inline gain/selector.py]: I don't think we need consider such a strange page.
 :joy: :joy: :joy:

Comment (hyfc) [inline gain/selector.py]: > To focus on titles or to focus on authors, that is the question.      ——not Shakespeare

If we decide to only focus on titles, then each titles can have an _auxiliary field_ to record zero or more  authors and vice versa. But it's too big a question for us to decide. 
Maybe consider something similar to the Object Relational Mapping paradigm, and let users decide which item is **primary key** and which is **foreign key**.

**Pull request closed without merging**